### PR TITLE
Use graphql-tag in a pragmatic way

### PR DESCRIPTION
**Pull request by xavxyz**
> Fixes issues #1, #2, #3.

🕹 Current state
---
The current state of `jest-transform-graphql` depends explicitly on an outdated version of `graphql-tag` _and_ run basically this version of `gql` on source files. 

🚧 Problem
---
We cannot use "evolved" graphql syntax in our files (such as imports) and are tied to an outdated version of `graphql-tag`.

✈️ Straightforward solution
---
Replace the `process` part of this transform by the actual code of `graphql-tag/loader`, such as shown in https://github.com/remind101/jest-transform-graphql/issues/2#issue-209120385 or proposed in https://github.com/remind101/jest-transform-graphql/pull/4.

This code works well and is about nearly the same as [`graphql-tag/loader`, except that we remove the `cacheable` option](https://github.com/apollographql/graphql-tag/blob/07c32a8c24bef51eba336625702b3063c1e314d7/loader.js#L26) used by Webpack. 

🚀 Pragmatic solution
---
> tldr; DRY. 

We can move the latest `graphql-tag` version as a peer dependency and require directly the loader from it. We then call the loader on the source file with a stubbed context so that the `cacheable` function exists.

We do not rewrite the graphql-tag code and leverages it at its best 😊 Thoughts? 💭

### CODE REVIEW

**Comment by turadg** [comment]
Awesome! Thanks for this contribution and thorough description.

**Comment by paradite** [comment]
Just a note that this PR creates some complexity for monorepos. For example, with the following setup:

Hoisted dependency (`./packages/`):
- jest-transform-graphql
  - ~graphql-tag~ (missing)

Package dependency (`./packages/abc/`):
- apollo-client
  - graphql-tag

jest would throw errors for missing `graphql-tag` since it cannot find `graphql-tag` in the hoisted directory `./packages/`.
This issue is masked in normal setup since other graphql libraries such as `apollo-client` has an dependency on `graphql-tag`, but using monorepo with hoisting reveals this issue.

I don't know if we should somehow fix it, given that the walkaround is pretty easy (declare `graphql-tag` separately as hoisted dependency).

**Comment by donedgardo** [comment]
This didn't seem to resolve importing fragment issue. #8 some help would be appreciated.
